src/provider/java/adoptiumJavaVersions.ts

```diff
@@ -0,0 +1,32 @@
+import { ADOPTIUM_API } from "#common/constants/urls.ts";
+import { moduleLogger } from "#core/logger.ts";
+import { defineProvider } from "#core/provider.ts";
+import { AdoptiumJavaReleases, AdoptiumJavaRuntimeEntries } from "#schema/java/adoptiumJavaData.ts";
+
+const RUNTIMES_URL = new URL("v3/", ADOPTIUM_API);
+
+const logger = moduleLogger();
+
+export default defineProvider({
+	id: "adoptium-java",
+
+	async provide(http): Promise<(AdoptiumJavaRuntimeEntries | null)[]> {
+		const releases = AdoptiumJavaReleases.parse(
+			(await http.getCached(new URL("info/available_releases", RUNTIMES_URL), "adoptium-java-available_releases.json")).json()
```

Comment:
Each provider gets its own folder so you don't need to make the name too elaborate.
```suggestion
			(await http.getCached(new URL("info/available_releases", RUNTIMES_URL), "available-releases.json")).json()
```